### ISSUE TITLE
QTextCursor::setPosition: Position '16786' out of range

### ISSUE THREAD

**Issue by aweinstein** [ipython/ipython#1392]
Sometimes (rather infrequently) when using the IPython QtConsole I get the following error message (in the terminal where I launched the QtConsole):

QTextCursor::setPosition: Position '16786' out of range

And an incorrect behavior with the prompt in the QtConsole itself. 

I am running Python 2.7.2, IPython 0.12 in an Ubuntu 11.10 machine.

This is a screenshot right after I got the error:

![Screenshot](https://lh3.googleusercontent.com/-mM-KZtBlX5I/TzNDQWZI_MI/AAAAAAAABi8/fSMrijmXfdk/s604/ipython_qtconsole.png)

**Comment by dashesy** [comment]
I get `QTextCursor::setPosition: Position '362277' out of range` and when it happens I have to close and re-open the window, restarting the kernel does not help.

**Comment by dashesy** [comment]
I realized when it hangs and prints this error to console, restarting the kernel will not help, but one can still type. If I type say `gg` and hit enter it does nothing but if I type `gg()` then hit enter it complains about invalid function but gives me a new line that can work again so it seems input processing is hanging and waiting for something.

**Comment by dashesy** [comment]
I noticed this happens almost reproducibly  after I use `?` to get `dir` of something like `np.unique?` then hit `q` to get to the prompt again.

**Comment by takluyver** [comment]
That didn't reproduce it for me on master (I tried several times).

**Comment by dashesy** [comment]
after upgrading from 3.0 to 3.2 I do not see this any more

**Comment by takluyver** [comment]
OK, let's close this for now - we can reopen if we find another reproducible case.

**Comment by jrovegno** [comment]
I could reproduce the problem, when i use qtconsole for a long time.
Jupyter QtConsole 4.1.1
Python 2.7.11 |Anaconda 2.4.1 (64-bit)| (default, Dec  7 2015, 14:10:42) [MSC v.1500 64 bit (AMD64)]

**Comment by slehar** [comment]
I have found a test case that reproduces the problem *almost* reliably, hopefully this might help us converge on the problem.

**Comment by dashesy** [comment]
@slehar can you also attach `BurrIndx.png`.

**Comment by dashesy** [comment]
Ok I renamed the png to `BurrIndx.png`. I tested it on Linux. While I do not see the `QTextCursor::setPosition: Position '13547' out of range` warnings any more I see this when I issue `ls` (but not `dir()`):

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib64/python2.7/lib-tk/Tkinter.py", line 1536, in __call__
    return self.func(*args)
  File "/usr/lib64/python2.7/site-packages/matplotlib/backends/backend_tkagg.py", line 383, in motion_notify_event
    FigureCanvasBase.motion_notify_event(self, x, y, guiEvent=event)
  File "/usr/lib64/python2.7/site-packages/matplotlib/backend_bases.py", line 1936, in motion_notify_event
    self.callbacks.process(s, event)
  File "/usr/lib64/python2.7/site-packages/matplotlib/cbook.py", line 540, in process
    proxy(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/matplotlib/cbook.py", line 415, in __call__
    return mtd(*args, **kwargs)
  File "/home/dashesy/Downloads/QTextCursorError.py", line 25, in onmotion
    print 'x=%3d, y=%3d, pixel=%3d'%(imgX, imgY, int(burrIndx[imgY, imgX]))
IndexError: index 25 is out of bounds for axis 0 with size 24

```

This is IPython 2.4.1 I am going to test on Jupyter now.

**Comment by dashesy** [comment]
Ok I see the errors with Jupyter on master branch so it is a good reproducible case:

```
QTextCursor::setPosition: Position '32400' out of range
```

    Jupyter QtConsole 4.3.0.dev
    IPython 4.1.1 
    Python 2.7.10 

If  one runs `%pylab` to have the image it occurs more, but even without `%pylab` it can be seen just click on few places outside the image. It happens any time the image is occluded with the terminal for example.

**Comment by slehar** [comment]
Sorry Dashesy, didn't notice your reply (shouldn't I get an email when people respond to my post?)
Yes I included BurrIndx.png beneath the attachment (tiny image for testing purposes)
My versions are:
Spyder 2.3.7 
Python 2.7.10 64bits, Qt 4.8.7, PyQt4 (API v2) 4.11.4 on Darwin
Let me know if I can be of further help.

**Comment by wmvanvliet** [comment]
I'm attempting a fix in  jupyter/qtconsole#145
